### ISSUE TITLE
Unable to evaluate the model

### ISSUE THREAD

**Issue by sglucas** [neelsjain/NEFTune#9]
Thank you very much for your great work!
I try to run the evaluation script `scripts/alpaca_eval.sh` and it occurs an error: 
```
File"hug/lib/python3.11/site-packages/accelerate/utils/modeling.py", line 1081, in load_checkpoint_in_model
    checkpoint_files = sorted(list(set(index.values())))
                                   ^^^^^^^^^^^^^^^^^^^
TypeError: unhashable type: 'list'
```
It could be better if you could help me fix it.

Thank you very much in advance!

Best
Lucas

**Comment by sglucas** [comment]
In addition,

```
MODEL_PATH=checkpoints/naive/1217/model
MODEL_NAME=llama7B_Alpaca
MODEL_CONFIG=<My local directory of config file from huggingface llama-7B-hf/config.json>
```

**Comment by neelsjain** [comment]
Can you provide the full stack of the error and the which versions of transformers, accelerate, and pytorch you used? I would double check the path checkpoints/naive/1217/model has the `.metadata` and the model as well. It seems that model cannot be loaded for some reason which makes me think that either (1) there is mismatch in the config vs model or (2) there is something wrong with the saved model. However, it is very difficult to tell without more information.

**Comment by sglucas** [comment]
Thanks for your reply.

I don't find `.metadata` in  checkpoints/naive/1217/model, I just find two files: `__0_0.distcp` and ` __1_0.distcp`

The configuration is attached: 

accelerate         0.20.3
aiohttp            3.8.6
aiosignal          1.3.1
alpaca-eval        0.3.1
appdirs            1.4.4
async-timeout      4.0.3
attrs              23.1.0
certifi            2023.7.22
charset-normalizer 3.3.0
click              8.1.7
cmake              3.27.7
datasets           2.12.0
dill               0.3.6
docker-pycreds     0.4.0
filelock           3.12.4
fire               0.5.0
frozenlist         1.4.0
fsspec             2023.10.0
gitdb              4.0.11
GitPython          3.1.40
huggingface-hub    0.18.0
idna               3.4
Jinja2             3.1.2
lion-pytorch       0.1.2
lit                17.0.3
MarkupSafe         2.1.3
mpmath             1.3.0
multidict          6.0.4
multiprocess       0.70.14
networkx           3.2
numpy              1.26.1
openai             0.27.8
packaging          23.2
pandas             2.1.1
pathtools          0.1.2
Pillow             10.1.0
pip                23.3
protobuf           3.20.0
psutil             5.9.6
pyarrow            13.0.0
python-dateutil    2.8.2
python-dotenv      1.0.0
pytz               2023.3.post1
PyYAML             6.0.1
regex              2023.10.3
requests           2.31.0
responses          0.18.0
safetensors        0.4.0
sentencepiece      0.1.99
sentry-sdk         1.32.0
setproctitle       1.3.3
setuptools         68.0.0
shortuuid          1.0.11
six                1.16.0
smmap              5.0.1
sympy              1.12
termcolor          2.3.0
tiktoken           0.5.1
tokenizers         0.13.3
torch              2.0.1+cu118
torchaudio         2.0.2+cu118
torchvision        0.15.2+cu118
tqdm               4.66.1
transformers       4.30.1
triton             2.0.0
typing_extensions  4.8.0
tzdata             2023.3
urllib3            2.0.7
wandb              0.15.12
wheel              0.41.2
xxhash             3.4.1
yarl               1.9.2

**Comment by neelsjain** [comment]
Ya, you might want to double check that the saving is being done correctly. It should contain a `.metadata` file. It is weird that it is not there. You can set max steps to like 1 to help with debugging this.

**Comment by aSingh071954** [comment]
I verified that my location has .metadata directory.
I am using the following command:-
MODEL_PATH=checkpoints/neftune/1217/model && MODEL_NAME=llama7B_Alpaca && MODEL_CONFIG=checkpoints_dum/neftune/1217/model/config.json && ANSWER_FILE=outputs/answers/alpaca_eval_${MODEL_NAME}.json && python eval_generate.py --model $MODEL_PATH --file_path alpaca_eval --save_file_name $ANSWER_FILE --model_name $MODEL_NAME --model_config_path $MODEL_CONFIG



===It seems it is trying to sort the list, which is not possible. see below output from config.json file
The model weights are not tied. Please use the `tie_weights` method before using the `infer_auto_device` function.
The model weights are not tied. Please use the `tie_weights` method before using the `infer_auto_device` function.
checkpoint_folder= checkpoints_dum/neftune/1217/model
index=, {'_name_or_path': 'meta-llama/Llama-2-7b-hf', 'architectures': ['LlamaForCausalLM'], 'bos_token_id': 1, 'do_sample': True, 'eos_token_id': 2, 'hidden_act': 'silu', 'hidden_size': 4096, 'initializer_range': 0.02, 'intermediate_size': 11008, 'max_length': 4096, 'max_position_embeddings': 2048, 'model_type': 'llama', 'num_attention_heads': 32, 'num_hidden_layers': 32, 'pad_token_id': 0, 'rms_norm_eps': 1e-06, 'temperature': 0.6, 'tie_word_embeddings': False, 'top_p': 0.9, 'torch_dtype': 'float32', 'transformers_version': '4.30.1', 'use_cache': True, 'vocab_size': 32001}
nat index=, dict_values(['meta-llama/Llama-2-7b-hf', ['LlamaForCausalLM'], 1, True, 2, 'silu', 4096, 0.02, 11008, 4096, 2048, 'llama', 32, 32, 0, 1e-06, 0.6, False, 0.9, 'float32', '4.30.1', True, 32001])

Traceback (most recent call last):
  File "/home/paperspace/exp/NEFTune/experiment_code/eval_generate.py", line 68, in <module>
    model = load_fsdp_ckpt_with_accelerate(args.model, model_config, hf_dummy_path=args.model_config_path, wrapped_class="LlamaDecoderLayer" if 'llama' in args.model else "OPTDecoderLayer")
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/paperspace/exp/NEFTune/experiment_code/utils.py", line 63, in load_fsdp_ckpt_with_accelerate
    model = load_checkpoint_and_dispatch(model_empty, hf_dummy_path, device_map="auto", no_split_module_classes=[wrapped_class, torch.nn.Embedding])
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/paperspace/miniconda3/envs/hug/lib/python3.11/site-packages/accelerate/big_modeling.py", line 486, in load_checkpoint_and_dispatch
    load_checkpoint_in_model(
  File "/home/paperspace/miniconda3/envs/hug/lib/python3.11/site-packages/accelerate/utils/modeling.py", line 1084, in load_checkpoint_in_model
    checkpoint_files = sorted(list(set(index.values())))
                                   ^^^^^^^^^^^^^^^^^^^

=====
So basically you can not create set out of list which is an item in the config.json file. What am i Missing, any help would be really appreciated ?

**Comment by neelsjain** [comment]
I would confirm that you are using the correct packages versions.

**Comment by neelsjain** [comment]
So, looking at the arg inputs from above I think you want to point to the directory where the config file is not the actual file. This should be where the original HF model is located.

**Comment by KuofengGao** [comment]
I have corrected the args inputs MODEL_CONFIG and successfully perform the evaluation.
Thank you very much for your reply which solves my problem.

> So, looking at the arg inputs from above I think you want to point to the directory where the config file is not the actual file. This should be where the original HF model is located.
